docs/pages/management/join-services-to-your-cluster/gcp.mdx

```diff
@@ -0,0 +1,101 @@
+---
+title: Join Services with GCP
+description: Use the GCP join method to add services to your Teleport cluster.
+---
+
+This guide will explain how to use the **GCP join method** to configure Teleport
+processes to join your Teleport cluster without sharing any secrets when they
+are running in a GCP VM.
+
+The GCP join method is available to any Teleport process running on a GCP VM.
+The VM must have a
+[service account](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances)
+assigned to it (the default service account is fine). No IAM roles are required
+on the Teleport process joining the cluster.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A GCP VM to host a Teleport service, with a service account assigned to it
+  and with the Teleport binary installed.
+- (!docs/pages/includes/tctl.mdx!)
+
+## Step 1/3. Create the GCP joining token
+
+Configure your Teleport Auth Service with a special dynamic token which will
+allow services from your GCP projects to join your Teleport cluster.
+
+Under the hood, services will prove that they are running in your GCP project
+by sending a signed ID token which matches an allow rule configured in your GCP
+joining token.
+
+Create the following `token.yaml` with a `gcp.allow` rule specifying your GCP
+project ID(s), service account(s), and location(s) in which your GCP instances
+will run.
+
+```yaml
+# token.yaml
+kind: token
+version: v2
+metadata:
+  # the token name is not a secret because instances must prove that they are
+  # running in your GCP project to use this token
+  name: gcp-token
+spec:
+  # use the minimal set of roles required (e.g. Node, App, Kube, DB, WindowsDesktop)
+  roles: [Node]
+
+  # set the join method allowed for this token
+  join_method: gcp
+
+  gcp:
+    allow:
+      # The GCP project ID(s) that VMs can join from.
+      - project_ids: ["example-project-id"]
+        # (Optional) The locations that VMs can join from. Note: both regions and
+        # zones are accepted.
+        locations: ["us-west1", "us-west2-a"]
+        # (Optional) The email addresses of service accounts that VMs can join
+        # with.
+        service_accounts: ["example@example.com"]
+```
+
+Run the following command to create the token:
+
+```code
+$ tctl create token.yaml
+```
+
+## Step 2/3. Configure your services
+
+The GCP join method can be used for Teleport processes running the SSH, Proxy,
```

Comment:
Since the SSH service is referred to in the token `spec.roles` as `Node`, should we hint to that here? Something like:

```suggestion
The GCP join method can be used for Teleport processes running the SSH (`Node`), Proxy,
```

Reply:
And should the proxy service be listed in the file comments on L46?